src/base_generator.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2021 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FLATBUFFERS_BASE_GENERATOR_H_
+#define FLATBUFFERS_BASE_GENERATOR_H_
+
+#include <cstdint>
+
+#include "flatbuffers/generator.h"
+#include "flatbuffers/reflection_generated.h"
+
+namespace flatbuffers {
+
+// A concrete base Flatbuffer Generator that specific language generators can
+// derive from.
+class BaseGenerator : public Generator {
```

Comment:
We already have a `BaseGenerator` class elsewhere.. even if we'd put it in a different namespace, recommend to rename both this class and the file it sits in avoid confusion. Not to mention ODR problems :)

Reply:
Done